File: 02-EstructurasDeControl/programasJava/Fruteria.java
package mx.edu.uttt.estructurasdecontrol;
import javax.swing.JOptionPane;

public class Fruteria {
    public static void main(String[] args) {
        double precioPorKilo, numKilos, subtotal, descuento, precioTotal;
        int resp;
        do {
            do {
                precioPorKilo = Double.parseDouble(JOptionPane.showInputDialog(null, "Ingrese el precio por kilo:", "Frutería", JOptionPane.QUESTION_MESSAGE));
                if (precioPorKilo <= 0) {
                    JOptionPane.showMessageDialog(null, "Debe ingresar datos válidos", "ERROR", JOptionPane.ERROR_MESSAGE);
                }
            } while (precioPorKilo <= 0);


            do {
                numKilos = Double.parseDouble(JOptionPane.showInputDialog(null, "Ingrese el número de kilos de fruta:", "Frutería", JOptionPane.QUESTION_MESSAGE));
                if (numKilos < 0) {
                    JOptionPane.showMessageDialog(null, "Debe ingresar datos válidos", "ERROR", JOptionPane.ERROR_MESSAGE);
                }
            } while (numKilos < 0);

            if (numKilos >= 0 && numKilos <= 2) {
                subtotal = precioPorKilo * numKilos;
                descuento = (subtotal * 0);
                precioTotal = subtotal - descuento;

                JOptionPane.showMessageDialog(null, "Número de kilos: " + numKilos + " kg\nPrecio por kilo:  $" + precioPorKilo + "\nSubtotal:  $" + subtotal + "\nDescuento:  $" + descuento + "\nTotal a pagar:  $" + precioTotal, "Frutería", JOptionPane.INFORMATION_MESSAGE);
            } else if (numKilos > 2 && numKilos <= 5) {
                subtotal = precioPorKilo * numKilos;
                descuento = (subtotal * 0.1);
                precioTotal = subtotal - descuento;

                JOptionPane.showMessageDialog(null, "Número de kilos: " + numKilos + " kg\nPrecio por kilo:  $" + precioPorKilo + "\nSubtotal:  $" + subtotal + "\nDescuento:  $" + descuento + "\nTotal a pagar:  $" + precioTotal, "Frutería", JOptionPane.INFORMATION_MESSAGE);
            } else if (numKilos > 5 && numKilos <= 10) {
                subtotal = precioPorKilo * numKilos;
                descuento = (subtotal * 0.15);
                precioTotal = subtotal - descuento;

                JOptionPane.showMessageDialog(null, "Número de kilos: " + numKilos + " kg\nPrecio por kilo:  $" + precioPorKilo + "\nSubtotal:  $" + subtotal + "\nDescuento:  $" + descuento + "\nTotal a pagar:  $" + precioTotal, "Frutería", JOptionPane.INFORMATION_MESSAGE);
            } else if (numKilos > 10) {
                subtotal = precioPorKilo * numKilos;
                descuento = (subtotal * 0.2);
                precioTotal = subtotal - descuento;

                JOptionPane.showMessageDialog(null, "Número de kilos: " + numKilos + " kg\nPrecio por kilo:  $" + precioPorKilo + "\nSubtotal:  $" + subtotal + "\nDescuento:  $" + descuento + "\nTotal a pagar:  $" + precioTotal, "Frutería", JOptionPane.INFORMATION_MESSAGE);
            }

            resp = JOptionPane.showConfirmDialog(null, "¿Desea repetir el programa?", "Frutería", JOptionPane.YES_NO_OPTION);
        } while (resp == 0);
    }
}

File: 02-EstructurasDeControl/programasPython/fruteria.py
# Declaración de variables
precio_por_kilo = 0.0
numero_kilos = 0.0
subtotal = 0.0
descuento = 0.0
precio_total = 0.0

# Entrada de datos 
precio_por_kilo = float(input("Ingrese el precio por kilo: "))
numero_kilos = float(input("Ingrese el numero de kilos de fruta: "))

if numero_kilos >= 0 & numero_kilos <= 2:    # 0 a 2 kg (0% de descuento)
    subtotal = precio_por_kilo * numero_kilos
    descuento = subtotal * 0
    precio_total = subtotal - descuento

    print(f"Número de kilos: {numero_kilos} kg\nPrecio por kilo: ${precio_por_kilo}\nSubtotal: ${subtotal}\nDescuento: ${descuento}\nTotal a pagar: ${precio_total}")
elif numero_kilos > 2 & numero_kilos <= 5:    # +2 a 5 kg (10% de descuento)
    subtotal = precio_por_kilo * numero_kilos
    descuento = subtotal * 0.1
    precio_total = subtotal - descuento

    print(f"Número de kilos: {numero_kilos} kg\nPrecio por kilo: ${precio_por_kilo}\nSubtotal: ${subtotal}\nDescuento: ${descuento}\nTotal a pagar: ${precio_total}")
elif numero_kilos > 5 & numero_kilos <= 10:    # +5 a 10 kg (15% de descuento)
    subtotal = precio_por_kilo * numero_kilos
    descuento = subtotal * 0.15
    precio_total = subtotal - descuento

    print(f"Número de kilos: {numero_kilos} kg\nPrecio por kilo: ${precio_por_kilo}\nSubtotal: ${subtotal}\nDescuento: ${descuento}\nTotal a pagar: ${precio_total}")
elif numero_kilos > 10:    # +10 kg (20% de descuento)
    subtotal = precio_por_kilo * numero_kilos
    descuento = subtotal * 0.2
    precio_total = subtotal - descuento

    print(f"Número de kilos: {numero_kilos} kg\nPrecio por kilo: ${precio_por_kilo}\nSubtotal: ${subtotal}\nDescuento: ${descuento}\nTotal a pagar: ${precio_total}")